src/time_integration/methods_2N.jl

```diff
@@ -129,10 +128,33 @@ function solve(ode::ODEProblem, alg::T;
         error("unsupported")
     end
 
-    solve!(integrator)
+    return integrator
+end
+
+# Fakes `solve`: https://diffeq.sciml.ai/v6.8/basics/overview/#Solving-the-Problems-1
+function solve(ode::ODEProblem, alg::T;
+               dt, callback = nothing, kwargs...) where {T <: SimpleAlgorithm2N}
+    integrator = init(ode, alg, dt = dt, callback = callback; kwargs...)
+
+    # Start actual solve
+    solve_steps!(integrator)
+end
+
+function solve_steps!(integrator::SimpleIntegrator2N)
+    @unpack prob = integrator.sol
+
+    integrator.finalstep = false
+
+    @trixi_timeit timer() "main loop" while !integrator.finalstep
+        step!(integrator)
+    end # "main loop" timer
+
+    return TimeIntegratorSolution((first(prob.tspan), integrator.t),
+                                  (prob.u0, integrator.u),
+                                  integrator.sol.prob)
 end
 
-function solve!(integrator::SimpleIntegrator2N)
+function step!(integrator::SimpleIntegrator2N)
```

Comment:
This deviates from OrdinaryDiffEq and is thus confusing, I think.

Reply:
So this is meant to be roughly 

https://github.com/SciML/OrdinaryDiffEq.jl/blob/6e9cf4055f8fc53f88af8a97df034d50927dd869/src/iterator_interface.jl

Do you think this is misleading/not equivalent enough to justify the same naming?

Reply:
As described at https://docs.sciml.ai/DiffEqDocs/latest/basics/integrator/#Initialization-and-Stepping, `step!(integrator)` takes one successful time integration step, e.g.,

```julia
julia> using OrdinaryDiffEq

julia> ode = ODEProblem((u, p, t) -> u, 1.0, (0.0, 1000.0))
ODEProblem with uType Float64 and tType Float64. In-place: false
timespan: (0.0, 1000.0)
u0: 1.0

julia> integrator = init(ode, Tsit5())
t: 0.0
u: 1.0

julia> step!(integrator)

julia> integrator
t: 0.10001999200479662
u: 1.105193012902056

julia> step!(integrator)

julia> integrator
t: 0.3480203186480749
u: 1.4162610240705922
```

However, this function here does all the `@trixi_timeit timer() "main loop" while !integrator.finalstep` stuff, so the full integration, doesn't it? Thus, it's basically `solve!(integrator)`, not `step!(integrator)` - or do I miss something?

Reply:
Yeah true - this is not correct. 

@warisa-r Please compare the proposed changes to the implementation for PERK2, i.e., https://github.com/trixi-framework/Trixi.jl/blob/5398b2257a68c8795be4ee06b5ee3e947e006ff8/src/time_integration/paired_explicit_runge_kutta/methods_PERK2.jl#L301-L313

Note in particular the placement of the loops.

Reply:
Copy that! I fixed it now. Sorry for this.